### PR TITLE
feat: Render the document shepherd writeup templates at two new URLs

### PR DESCRIPTION
**Pull request by larseggert**
Those being `/doc/shepherdwriteup-template/group` and `/doc/shepherdwriteup-template/individual`.

CC @ghwood.

### CODE REVIEW

**Review by jennifer-richards — APPROVED** [review]
A couple comments inline - the awkward Boolean logic in the template bugs me a bit. Maybe consider always passing `type` in the context instead of adding the complication of looking both at `type` and `doc`?

Also, I think this is the only place where the writeup is rendered with markdown - is that intentional? (I don't think it's rendered in the `document_shepherd_writeup()` view yet, but maybe I missed it?

These are just for consideration, no objection to running with it as-is.

File: ietf/utils/markdown.py [hunk 1]
```diff
@@ -8,20 +8,17 @@
...
 
 def markdown(text):
     return mark_safe(
         bleach_linker.linkify(
             urlize_ietf_docs(
-                bleach_cleaner.clean(
-                    python_markdown.markdown(
-                        text, extensions=[ExtraExtension(), "nl2br"]
-                    )
+                python_markdown.markdown(
```
[the remark this answers is not in the document]
Comment (larseggert) [reply ietf/utils/markdown.py:20]: I will add it back, but we need to check that it doesn't strip attributes we'd want in markdown output (like stripping `start` from `ol`).

**Comment by larseggert** [comment]
I restored the `bleach` step

**Review by rjsparks — APPROVED** [review]

**Comment by larseggert** [comment]
Do you _want_ it in another PR, or can I add it to this one?

**Comment by jennifer-richards** [comment]
I'm worried about whether bleaching the input leads to a false sense of security. I haven't found recent discussions, but there seems to be a general sense that feeding markdown into bleach is not valid. I think we should probably stick to teaching bleach to allow the things we need or we're likely baking in vulnerabilities.

**Comment by rjsparks** [comment]
I think adding it to this one is fine. Could you and Jennifer iterate on whether we're doing the right thing with bleach at the right place?

**Comment by jennifer-richards** [comment]
@larseggert, Were there particular pieces of markdown formatting that bleach was still stripping?

**Comment by jennifer-richards** [comment]
Just looked - I think the "id" attributes were being stripped from the rendered HTML:
```
$ diff bleached-individual.html unbleached-individual.html 
616c616
<     <h1>Document Shepherd Write-Up for Individual Documents</h1>
---
>     <h1 id="document-shepherd-write-up-for-individual-documents">Document Shepherd Write-Up for Individual Documents</h1>
626c626
< <h2>Document History</h2>
---
> <h2 id="document-history">Document History</h2>
650c650
< <h2>Additional Reviews</h2>
---
> <h2 id="additional-reviews">Additional Reviews</h2>
676c676
< <h2>Document Shepherd Checks</h2>
---
> <h2 id="document-shepherd-checks">Document Shepherd Checks</h2>
```
and similar for the group template. Adding the "id" attribute to the allowed attributes list in `ietf/utils/text.py` (`attributes["*"] = ["id"]`, same as is done for the "start" attribute for "ol") fixes that and reduces the diff to null.

**Comment by larseggert** [comment]
Good catch. I'll see if I can cook up a test to see if there are others.

**Comment by larseggert** [comment]
There weren't any others, so I think this is good to go.

**Pull request merged**